src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1852,10 +1852,16 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
 
             if (this.expressionMap.eagerRelations === true) {
                 if (!this.findOptions.options || this.findOptions.options.eagerRelations !== false) {
+                    // Create a list of all of the alias+propertyPaths that were manually joined, so we don't join them again
+                    const manuallyJoinedRelations = this.expressionMap.joinAttributes.filter(join => join.relationPropertyPath).map(join => join.parentAlias + "." + join.relationPropertyPath);
```

Comment:
Line is pretty long. Maybe it would look better written like that:
```typescript
    const manuallyJoinedRelations = this.expressionMap.joinAttributes
                                        .filter(join => join.relationPropertyPath)
                                        .map(join => join.parentAlias + "." + join.relationPropertyPath);
```